### PR TITLE
go-neb: Init at unstable-2020-04-09

### PR DESCRIPTION
**Pull request by maralorn**
###### Motivation for this change

Adds a package for the go-neb matrix bot.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by mweinelt** [comment]
Result of `nixpkgs-review pr 87854` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
<br>- go-neb
</details>

**Review by mweinelt — APPROVED** [review]

**Comment by Gaelan** [comment]
Builds successfully on Darwin.

**Comment by mweinelt** [comment]
This pr is currently rebased on #87855 and should be merged after that.

**Comment by maralorn** [comment]
I have reverted the dependency on #87855 because that PR got rejected. It now uses the old patch method for the modfile again.

**Comment by mweinelt** [comment]
Result of `nixpkgs-review pr 87854` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
<br>- go-neb
</details>

**Review by bbigras — APPROVED** [review]

**Comment by bbigras** [comment]
Result of `nixpkgs-review pr 87854` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
<br>- go-neb
</details>

**Comment by bbigras** [comment]
Builds and runs fine. Tested with a real notification from alertmanager/prometheus to a room on matrix.org. Thanks for the PR!

File: pkgs/applications/networking/instant-messengers/go-neb/default.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  patches = [ ./go-mod.patch ];
+
+  vendorSha256 = "1k3980yf6zl00dkd1djwhm2f9nnffzrsbs3kq3alpw2gm0aln739";
+
+  meta = with lib; {
+    description = "Extensible matrix bot written in Go";
+    homepage = "https://github.com/matrix-org/go-neb";
+    license = licenses.asl20;
+    maintainers = with maintainers; [ hexa maralorn ];
+    platforms = platforms.linux ++ platforms.darwin;
```
Comment (kalbasit) [inline pkgs/applications/networking/instant-messengers/go-neb/default.nix:24]:
```suggestion
```

It's already set to this value:
https://github.com/NixOS/nixpkgs/blob/765e5e5128c49026a798e85eba9e58b04440e577/pkgs/development/compilers/go/1.14.nix#L242

**Review by kalbasit — CHANGES_REQUESTED** [review]
Otherwise LGTM. Please address my comment and I can merge it

**Comment by maralorn** [comment]
@kalbasit Done

**Review by kalbasit — APPROVED** [review]

**Comment by kalbasit** [comment]
@GrahamcOfBorg build go-neb

**Comment by bbigras** [comment]
@maralorn any plans for a nixos module?

**Comment by mweinelt** [comment]
@bbigras Yep, we intend to write that tonight(TM).

**Comment by bbigras** [comment]
Sweet, thanks!

**Pull request merged**